### PR TITLE
Debug mode can be turned on/off via --debug argument

### PR DESCRIPTION
**Pull request by Cropi**
As default, log messages are turned off. You might want to turn the on using the **--debug** argument.

### CODE REVIEW

File: src/Log.cpp [hunk 1]
```diff
@@ -45,4 +48,15 @@ LoggerStream Logger::createLogMessage(const std::string& file, const std::string
     };
     return LoggerStream(*this, source);
 }
+
+void Logger::setDebugMode(bool debug)
+{
+    _debug = debug;
+}
+
+bool Logger::isEnabled()
```
Comment (ZoltanFridrich) [inline src/Log.cpp]: Please start using C++ features like const and noexcept where it is appropriate like exactly here

File: src/Log.cpp [hunk 2]
```diff
@@ -45,4 +48,15 @@ LoggerStream Logger::createLogMessage(const std::string& file, const std::string
     };
     return LoggerStream(*this, source);
 }
+
+void Logger::setDebugMode(bool debug)
```
Comment (ZoltanFridrich) [inline src/Log.cpp]: another noexcept contender

File: src/Log.hpp [hunk 3]
```diff
@@ -12,8 +12,11 @@ class LoggerStream;
 class Logger
 {
 public:
-    Logger();
+    Logger(bool debug = false);
     LoggerStream createLogMessage(const std::string& file, const std::string& function, int line);
+    void setDebugMode(bool debug);
+    bool isEnabled();
+    bool _debug;
```
Comment (ZoltanFridrich) [inline src/Log.hpp]: why is this public?

File: src/Log.hpp [hunk 4]
```diff
@@ -38,8 +41,11 @@ class LoggerStream : public std::ostringstream
 
 extern Logger g_nLog;
 
+#define NOTIFIER_LOGGER usbguardNotifier::g_nLog
```
Comment (ZoltanFridrich) [inline src/Log.hpp]: If possible I would prefer not using define here and maybe use something else... maybe keyword "using" might do the trick

File: src/Log.hpp [hunk 5]
```diff
@@ -38,8 +41,11 @@ class LoggerStream : public std::ostringstream
...
+
 #define NOTIFIER_LOG() \
```
Comment (ZoltanFridrich) [inline src/Log.hpp]: make this a function maybe

File: src/Main.cpp [hunk 6]
```diff
@@ -19,13 +21,14 @@ void showHelp(const std::string& appName, std::ostream& output)
...
     output << "    -w, --wait      Wait until an active IPC connection is estabilished." << std::endl;
+    output << "    -d, --debug     Enable debug mode." << std::endl;
     output << "    -h, --help      Show this usage message." << std::endl;
 }
 
 int main(int argc, char** argv)
 {
     const std::string appName(*argv);
-    bool waitConnection = false;
+    bool waitConnection = false, debug = false;
```
Comment (ZoltanFridrich) [inline src/Main.cpp]: I personally would split this into two lines

File: src/Log.hpp (hunk 3 above)
Comment (Cropi) [reply src/Log.hpp]: ``setDebug()`` and ``isEnabled()`` are called from main. The can not be private within this context. However, ``_debug`` should be.

File: src/Log.hpp (hunk 4 above)
Comment (Cropi) [reply src/Log.hpp]: **using** and **define** are not the same. Defines are not program statements but directives for the preprocessor.

File: src/Log.hpp (hunk 5 above)
Comment (Cropi) [reply src/Log.hpp]: What do you mean by this?

Comment (ZoltanFridrich) [reply src/Log.hpp]: NOTIFIER_LOG() ... why is this not a function

File: src/Log.hpp (hunk 4 above)
Comment (ZoltanFridrich) [reply src/Log.hpp]: I teach C programming you dont have to tell me that Xd . I just dislike using defines in C++ when there are better options for sure

File: src/Log.hpp (hunk 3 above)
Comment (ZoltanFridrich) [reply src/Log.hpp]: I meant only _debug

File: src/Log.hpp (hunk 5 above)
Comment (Cropi) [reply src/Log.hpp]: Next time please mark the correct line. It's better to let``NOTIFIER_LOG``  stay like this for the following reason: we do want to know from which file, function and line did we inkove the logger. If you add this part to a function, you will get the same results whenever you call it.

Comment (ZoltanFridrich) [reply src/Log.hpp]: oh ofc I forgot about that

File: src/Log.hpp (hunk 4 above)
Comment (Cropi) [reply src/Log.hpp]: usbguardNotifier::g_nLog is not a type, so in my opinion you can not use ``using`` for this purpose. However, if you have a better solution for this, I would like to know about that. Thanks.

File: src/Log.cpp (hunk 1 above)
Comment (Cropi) [reply src/Log.cpp]: Okay

**Pull request merged**